src/kani-compiler/rustc_codegen_kani/src/codegen/statement.rs

```diff
@@ -78,18 +79,24 @@ impl<'tcx> GotocCtx<'tcx> {
                     if *expected { r } else { Expr::not(r) }
                 };
 
-                Stmt::block(
-                    vec![
-                        cond.cast_to(Type::bool()).if_then_else(
-                            Stmt::goto(self.current_fn().find_label(target), loc.clone()),
-                            None,
-                            loc.clone(),
-                        ),
-                        Stmt::assert_false(&format!("{:?}", msg), loc.clone()),
+                let mut msg_str = format!("{:?}", msg);
+                let mut stmts: Vec<Stmt> = Vec::new();
+                if self.queries.get_check_assertion_reachability() {
+                    let check_id = self.next_check_id();
+                    msg_str = format!("[{}] {}", check_id, msg_str);
+                    let reach_msg = GotocCtx::reach_check_msg(&check_id);
```

Comment:
Why is one of these behind a function and the other not?

Reply:
Good point! I added another function.